File: Programmers/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] end = new int[100];
        int day = -1;

        for(int i = 0; i < progresses.length; i++) {
            while(progresses[i] + (day * speeds[i]) < 100) {
                day++;
            }

            end[day]++;
        }
        
        return Arrays.stream(end).filter(i -> i != 0).toArray();
    }
}

File: Programmers/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> q;
    
    for (int i = 0; i < progresses.size(); i++)
    {
        int remain = 100 - progresses[i];
        int days = (remain + speeds[i]  - 1) / speeds[i];
        
        q.push(days);
    }
    
    while (!q.empty())
    {
        int releaseDay = q.front();
        q.pop();
        int count = 1;
        
        while (!q.empty() && q.front() <= releaseDay)
        {
            q.pop();
            count++;
        }
        
        answer.push_back(count);
    }
    
    return answer;
}